libs/wingsdk/src/cloud/function.ts

```diff
@@ -30,6 +30,12 @@ export interface FunctionProps {
    * @default 1m
    */
   readonly timeout?: Duration;
+
+  /**
+   * The amount of memory to allocate to the function, in MB.
+   * @default 128
+   */
+  readonly memory?: number;
```

Comment:
@raywonkari We should also implement or handle this field for other cloud targets. (What happens if `memory` is configured while you run your app on the Wing simulator or on Azure?)

- For the simulator, I think it is OK if we do not simulate memory limits since it would require significant rearchitecture, so we should just add a comment in the constructor [here](https://github.com/winglang/wing/blob/main/libs/wingsdk/src/target-sim/function.ts) saying that "props.memory is unused since we are not simulating it"
- For Azure, I don' t think the memory is directly configurable so I would throw an error in the constructor [here](https://github.com/winglang/wing/blob/main/libs/wingsdk/src/target-tf-azure/function.ts) if `props.memory` is configured (if the value is not undefined). cc @hasanaburayyan Let me know if this sounds ok